### PR TITLE
Add MATLAB unit tests

### LINKED ISSUE #56: MATLAB unit tests
Some utility functions in the [matlab/](https://github.com/UCL/TDMS/tree/main/tdms/matlab) directory are currently unused. Unit tests should be added to test the functionality. Specifically,

- fdtdduration.m
- minsteps_fdtd.m
- minsteps_pstd.m
- fdtdts.m

***
### Dependencies 

- #37 

***
### Starting point
(from @prmunro)

```matlab
%exports the constants used internally in the PSTD/FDTD
%code when using pulsed illumination
[to hwhm] = fdtdduration('pstd_input_file_2D_01.m');

%calculates the minimum number of time steps required for light to
%make one double pass of the computational domain. If using the
%FDTD algorithm, the script fdtdminsteps.m should be used
Nt = minsteps_pstd('pstd_input_file_2D_01.m');

%the maximum allowable timestep
dt_upper = fdtdts('pstd_input_file_2D_01.m');
```

### PR DESCRIPTION
**Pull request by t-young31**
Closes #56

### CODE REVIEW

File: tdms/matlab/fdtdduration.m [hunk 1]
```diff
@@ -6,34 +6,35 @@
...
-omega_an         = 2*pi*f_an;
-lambda_an        = c/(f_an*refractive_index);
+    refractive_index = sqrt(epsr(1));
+    omega_an         = 2*pi*f_an;
+    lambda_an        = c/(f_an*refractive_index);
 
-hwhm = lambda_an^2/((c/refractive_index)*wavelengthwidth)*2*sqrt(log(2)/pi);
-to   = hwhm*sqrt(log(1e8)/pi);
+    hwhm = lambda_an^2/((c/refractive_index)*wavelengthwidth)*2*sqrt(log(2)/pi);
+    to   = hwhm*sqrt(log(1e8)/pi);
```
Comment (t-young31) [inline tdms/matlab/fdtdduration.m:39]: @prmunro would you mind clarifying what this variable is? Thanks!

File: tdms/tests/matlab/test_fdtdduration.m [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    % Check that these are the expected values
+    verifyEqual(testCase, 3.035046893364502e-14, hwhm_actual, "AbsTol", 1e-10);
+    verifyEqual(testCase, 7.349254840150354e-14, to_actual, "AbsTol", 1e-10);
+end
+
+
+function testMinStepsFDTD(testCase)
+
+    n_expected = 1483;
+    % This also tests fdtdts, which computes the upper limite of the timestep
```
Comment (willGraham01) [inline tdms/tests/matlab/test_fdtdduration.m:23]:
```suggestion
    % This also tests fdtdts, which computes the upper limit of the number of timesteps
```

**Review by willGraham01 — APPROVED** [review]

File: tdms/matlab/fdtdduration.m (hunk 1 above)
Comment (prmunro) [reply tdms/matlab/fdtdduration.m]: to (which I should have called t0) controls the time delay of the pulse that enters the FDTD/PSTD simulation when a source is introduced. In particular, it ensures that when the pulse starts to be introduced, the envelope of the pulse has a magnitude of 10^(-8). This ensures that there aren't any erroneous spectral components introduced as a result of having a sharp discontinuity in time.

Comment (t-young31) [reply tdms/matlab/fdtdduration.m]: thanks – I've added a comment above the function. Will merge this PR once it has your approval

Comment (t-young31) [reply tdms/matlab/fdtdduration.m]: will merge this now with a view to not have too many long-running PRs. can come back to this when people are back from holiday!

**Pull request merged**